public/app/features/alerting/unified/components/rules/RulesTable.tsx

```diff
@@ -63,9 +63,19 @@ export const RulesTable = ({
 
   const { pageItems, page, numberOfPages, onPageChange } = usePagination(rules, 1, DEFAULT_PER_PAGE_PAGINATION);
 
-  const { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus } = useLazyLoadRulerRules(pageItems);
+  const [lazyLoadRules, { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus }] =
+    useLazyLoadRulerRules(pageItems);
+  const isLoadingRulerGroup = useMemo(
+    () =>
+      !rulerRulesLoadingStatus || rulerRulesLoadingStatus === 'loading' || rulerRulesLoadingStatus === 'not-executed',
+    [rulerRulesLoadingStatus]
+  );
 
-  const isLoadingRulerGroup = rulerRulesLoadingStatus === 'loading';
+  useEffect(() => {
+    if (pageItems.length > 0) {
+      lazyLoadRules.execute();
+    }
+  }, [lazyLoadRules, pageItems, rulerRulesLoadingStatus]);
```

Comment:
Moving the `execute` into here results in the state list rendering properly on initial page load. But there's a lot to unravel outside of this, and the logic/prop drilling that needs to happen to get to this point is quite complicated. 
I think overall, we're best revisiting/fixing this all once the new list view is closer to completion